indico/modules/events/abstracts/placeholders.py

```diff
@@ -105,9 +105,16 @@ class ContributionTypePlaceholder(Placeholder):
 
     @classmethod
     def render(cls, abstract):
-        if abstract.state == AbstractState.accepted:
+        if abstract.state == AbstractState.withdrawn:
+            if abstract.accepted_contrib_type:
+                return abstract.accepted_contrib_type.name
+            elif abstract.submitted_contrib_type:
+                return abstract.submitted_contrib_type.name
+            return ''
+        elif abstract.state != AbstractState.accepted:
+            return abstract.submitted_contrib_type.name if abstract.submitted_contrib_type else ''
```

Comment:
you can simplify the whole thing if you set `ctype = ...` and then at the very end do `return ctype.name if ctype else ''`